### PR TITLE
Stop creating duplicate PLP entries on import

### PR DESCRIPTION
**Pull request by aoby**

### CODE REVIEW

**Comment by ashercodeorg** [comment]
LGTM. Should there be a DB level constraint to prevent this from recurring?

**Comment by aoby** [comment]
Yeah, not a bad idea. I'll add a migration...

File: dashboard/db/migrate/20160913212311_add_unique_constraint_to_professional_learning_partners.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class AddUniqueConstraintToProfessionalLearningPartners < ActiveRecord::Migration[5.0]
+  def change
+    change_table :professional_learning_partners do |t|
+      t.remove_index column: :name
+      t.index :name, unique: true
```
Comment (ashercodeorg) [inline dashboard/db/migrate/20160913212311_add_unique_constraint_to_professional_learning_partners.rb]: Should the uniqueness be on `name` or `(name, contact_id)` or `contact_id`?

Comment (ashercodeorg) [inline dashboard/db/migrate/20160913212311_add_unique_constraint_to_professional_learning_partners.rb]: Note that production currently violates the uniqueness on name.

```
SELECT cnt, COUNT(0) FROM (SELECT name, COUNT(0) cnt FROM professional_learning_partners GROUP BY 1) t GROUP BY 1;
+-----+----------+
| cnt | COUNT(0) |
+-----+----------+
|   1 |       16 |
|   2 |        7 |
|   3 |        2 |
+-----+----------+
```

**Comment by ashercodeorg** [comment]
The migration should be run after the code change, so that things don't blow up if the old code gets executed between the last data fix and the migration.

File: dashboard/db/migrate/20160913212311_add_unique_constraint_to_professional_learning_partners.rb (hunk 1 above)
Comment (aoby) [reply dashboard/db/migrate/20160913212311_add_unique_constraint_to_professional_learning_partners.rb]: Yeah, you're right. And I just confirmed on the source sheet that some PLPs have multiple entries, one for each contact.

**Comment by aoby** [comment]
I updated the unique constraint to name and contact_id.

As we discussed offline, I'm not splitting the migration into a separate PR. The import script is run manually when needed so it won't execute on its own before the migration. I verified the `[:name, :contact_id]` unique constraint is safe on production, test, and staging via:

```
SELECT name, contact_id, COUNT(0) FROM professional_learning_partners GROUP BY name, contact_id HAVING COUNT(0) > 1;
```

**Comment by ashercodeorg** [comment]
SGTM.

**Pull request merged**